provider/core/src/hello_world.rs

```diff
@@ -268,18 +294,6 @@ impl<'l> Writeable for FormattedHelloWorld<'l> {
 
 writeable::impl_display_with_writeable!(FormattedHelloWorld<'_>);
 
-#[cfg(feature = "datagen")]
```

Comment:
Nit: As a general rule in PRs, please favor smaller diffs over moving code around

Reply:
Yes but this is somehow ended up with the formatter code.

Reply:
I think logically it is below the formatter code and above the test code. Also note that its unit test function is right below.

Reply:
It adds implementations on the data provider, so it should be with the other data provider implementations.